homeassistant/components/gotify/__init__.py

```diff
@@ -0,0 +1,28 @@
+"""The gotify integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up gotify from a config entry."""
+    hass.data.setdefault(DOMAIN, {})[entry.data[CONF_NAME]] = entry
```

Comment:
We usually store integration data as `hass.data.setdefault(DOMAIN, {})[entry.entry_id]`